opentracing-spring-cloud-tracerresolver/src/main/java/io/opentracing/contrib/spring/cloud/tracerresolver/TracerResolverConfiguration.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.tracerresolver;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import io.opentracing.contrib.tracerresolver.TracerResolver;
+
+@Configuration
+@ConditionalOnClass(value = {TracerResolver.class})
```

Comment:
Although it should probably be conditional on no `Tracer` bean as well, in case one has been provided.